src/rpc/rawtransaction.cpp

```diff
@@ -1107,10 +1157,61 @@ static RPCHelpMan decodepsbt()
 
     UniValue result(UniValue::VOBJ);
 
-    // Add the decoded tx
-    UniValue tx_univ(UniValue::VOBJ);
-    TxToUniv(CTransaction(*psbtx.tx), uint256(), tx_univ, false);
-    result.pushKV("tx", tx_univ);
+    if (psbtx.tx != std::nullopt) {
+        // Add the decoded tx
+        UniValue tx_univ(UniValue::VOBJ);
+        TxToUniv(CTransaction(*psbtx.tx), uint256(), tx_univ, false);
+        result.pushKV("tx", tx_univ);
+    }
+
+    // Add the global xpubs
+    UniValue global_xpubs(UniValue::VARR);
+    for (std::pair<KeyOriginInfo, std::set<CExtPubKey>> xpub_pair : psbtx.m_xpubs) {
+        for (auto& xpub : xpub_pair.second) {
+            std::vector<unsigned char> ser_xpub;
+            ser_xpub.assign(BIP32_EXTKEY_WITH_VERSION_SIZE, 0);
+            xpub.EncodeWithVersion(ser_xpub.data());
+
+            UniValue keypath(UniValue::VOBJ);
+            keypath.pushKV("xpub", EncodeBase58Check(ser_xpub));
+            keypath.pushKV("master_fingerprint", HexStr(Span<unsigned char>(xpub_pair.first.fingerprint, xpub_pair.first.fingerprint + 4)));
+            keypath.pushKV("path", WriteHDKeypath(xpub_pair.first.path));
+            global_xpubs.push_back(keypath);
+        }
+    }
+    result.pushKV("global_xpubs", global_xpubs);
+
+    // Add PSBTv2 stuff
+    if (psbtx.GetVersion() == 2) {
+        if (psbtx.tx_version != std::nullopt) {
+            result.pushKV("tx_version", *psbtx.tx_version);
+        }
+        if (psbtx.fallback_locktime != std::nullopt) {
+            result.pushKV("fallback_locktime", static_cast<uint64_t>(*psbtx.fallback_locktime));
+        }
+        result.pushKV("input_count", psbtx.inputs.size());
+        result.pushKV("output_count", psbtx.inputs.size());
```

Comment:
Getting a compile error consistent with the CI for these two lines:

```
rpc/rawtransaction.cpp:1192:16: error: call to member function 'pushKV' is ambiguous
        result.pushKV("input_count", psbtx.inputs.size());
        ~~~~~~~^~~~~~
./univalue/include/univalue.h:127:10: note: candidate function
    bool pushKV(const std::string& key, int64_t val_) {
         ^
./univalue/include/univalue.h:131:10: note: candidate function
    bool pushKV(const std::string& key, uint64_t val_) {
         ^
./univalue/include/univalue.h:135:10: note: candidate function
    bool pushKV(const std::string& key, bool val_) {
         ^
./univalue/include/univalue.h:139:10: note: candidate function
    bool pushKV(const std::string& key, int val_) {
         ^
./univalue/include/univalue.h:143:10: note: candidate function
    bool pushKV(const std::string& key, double val_) {
         ^
./univalue/include/univalue.h:118:10: note: candidate function
    bool pushKV(const std::string& key, const UniValue& val);
         ^
./univalue/include/univalue.h:119:10: note: candidate function not viable: no known conversion from 'std::__1::vector<PSBTInput, std::__1::allocator<PSBTInput> >::size_type' (aka 'unsigned long') to 'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char> >') for 2nd argument
    bool pushKV(const std::string& key, const std::string& val_) {
         ^
./univalue/include/univalue.h:123:10: note: candidate function not viable: no known conversion from 'std::__1::vector<PSBTInput, std::__1::allocator<PSBTInput> >::size_type' (aka 'unsigned long') to 'const char *' for 2nd argument
    bool pushKV(const std::string& key, const char *val_) {
         ^
rpc/rawtransaction.cpp:1193:16: error: call to member function 'pushKV' is ambiguous
        result.pushKV("output_count", psbtx.inputs.size());
        ~~~~~~~^~~~~~
./univalue/include/univalue.h:127:10: note: candidate function
    bool pushKV(const std::string& key, int64_t val_) {
         ^
./univalue/include/univalue.h:131:10: note: candidate function
    bool pushKV(const std::string& key, uint64_t val_) {
         ^
./univalue/include/univalue.h:135:10: note: candidate function
    bool pushKV(const std::string& key, bool val_) {
         ^
./univalue/include/univalue.h:139:10: note: candidate function
    bool pushKV(const std::string& key, int val_) {
         ^
./univalue/include/univalue.h:143:10: note: candidate function
    bool pushKV(const std::string& key, double val_) {
         ^
./univalue/include/univalue.h:118:10: note: candidate function
    bool pushKV(const std::string& key, const UniValue& val);
         ^
./univalue/include/univalue.h:119:10: note: candidate function not viable: no known conversion from 'std::__1::vector<PSBTInput, std::__1::allocator<PSBTInput> >::size_type' (aka 'unsigned long') to 'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char> >') for 2nd argument
    bool pushKV(const std::string& key, const std::string& val_) {
         ^
./univalue/include/univalue.h:123:10: note: candidate function not viable: no known conversion from 'std::__1::vector<PSBTInput, std::__1::allocator<PSBTInput> >::size_type' (aka 'unsigned long') to 'const char *' for 2nd argument
    bool pushKV(const std::string& key, const char *val_) {
         ^
2 errors generated.
```

Reply:
I can't replicate this error at all, but I think I have fixed it.